lib/info.dart

```diff
@@ -46,8 +50,26 @@ abstract class Info {
 // TODO(sigmund): add more:
 //  - inputSize: bytes used in the Dart source program
 abstract class BasicInfo implements Info {
+  static Set<int> _ids = new Set<int>();
   final InfoKind kind;
-  final int id;
+
+  int _id;
+  int get id {
+    if (_id == null) {
+      // we hope we have all of the parents set up by now...
+
+      var path = _getParentPath(this);
```

Comment:
But then that helper would only have one user, right?

I'd rather keep this private to the impl.